www/js/app.js

```diff
@@ -593,52 +645,60 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
                 setContentInjectionMode('jquery');
                 return;
             }
-            
+            var protocol = window.location.protocol;
             if (!isServiceWorkerReady()) {
                 $('#serviceWorkerStatus').html("ServiceWorker API available : trying to register it...");
-                navigator.serviceWorker.register('../service-worker.js').then(function (reg) {
-                    // The ServiceWorker is registered
-                    serviceWorkerRegistration = reg;
+                if (navigator.serviceWorker.controller) {
+                    console.log("Active service worker found, no need to register");
+                    serviceWorkerRegistration = true;
+                    // Remove any jQuery hooks from a previous jQuery session
+                    $('#articleContent').contents().remove();
+                    // Create the MessageChannel and send 'init'
+                    initOrKeepAliveServiceWorker();
                     refreshAPIStatus();
-                    
-                    // We need to wait for the ServiceWorker to be activated
-                    // before sending the first init message
-                    var serviceWorker = reg.installing || reg.waiting || reg.active;
-                    serviceWorker.addEventListener('statechange', function(statechangeevent) {
-                        if (statechangeevent.target.state === 'activated') {
-                            // Remove any jQuery hooks from a previous jQuery session
-                            $('#articleContent').contents().remove();
-                            // Create the MessageChannel
+                } else {
+                    if (protocol === 'moz-extension:') {
```

Comment:
Actually I originally ran `launchMozillaExtensionServiceWorker()` from the `catch` function of the Promise (around line 690 below, probably shifted once I push review changes). However, this was causing me problems, including timing problems, so I shifted it here. I could certainly try again putting it in the `catch`, because some of the timing issues I was having were later resolved by different code, so maybe it would work cleanly now.

Reply:
Now done.